### PR TITLE
removing not needed demos + migrating to 5.0

### PR DESCRIPTION
**Pull request by slavagithub**
removing not needed demos + migrating to 5.0.
JetInstanse things are still them the code, we'll remove them once Hazelcast.bootstrappedInstance() is available

### CODE REVIEW

File: cryptocurrency-sentiment-analysis/README.md [hunk 1]
```diff
@@ -1,66 +0,0 @@
-# Twitter Cryptocurrency Sentiment Analysis
```
Comment (olukas) [inline cryptocurrency-sentiment-analysis/README.md]: This demo is not supposed to be deleted!

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java [hunk 2]
```diff
@@ -163,16 +165,17 @@ public static void main(String[] args) {
             Job job = jet.newJob(pipeline, new JobConfig().setName("FlightTelemetry").setProcessingGuarantee(ProcessingGuarantee.EXACTLY_ONCE));
             job.join();
         } finally {
-            Jet.shutdownAll();
-        }
+            Hazelcast.shutdownAll();        }
```
Comment (olukas) [inline flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java:165]: Move `}` to another line

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java [hunk 3]
```diff
@@ -163,16 +165,17 @@ public static void main(String[] args) {
...
 
     private static JetInstance getJetInstance() {
         String bootstrap = System.getProperty("bootstrap");
         if (bootstrap != null && bootstrap.equals("true")) {
             return JetBootstrap.getInstance();
         }
-        return Jet.newJetInstance();
+
+//        return Jet.bootstrappedInstance().getHazelcastInstance();
+        return Jet.bootstrappedInstance();
```
Comment (olukas) [inline flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java]: This makes no sense to me - boostrap instance should be used if this `bootstrap != null && bootstrap.equals("true")` is true (previous condition). I guess migration of `Jet.newJetInstance()` should not be `Jet.bootstrappedInstance()` but `Hazelcast.newHazelcastInstance().getJetInstance()`. 

The same should be applied for the other replacements `Jet.newJetInstance()` by `Jet.bootstrappedInstance()` in this PR.

Also please remove the commented code.

File: pom.xml [hunk 4]
```diff
@@ -49,7 +47,7 @@
 
     <properties>
         <jdk.version>1.8</jdk.version>
-        <hazelcast-jet.version>4.3</hazelcast-jet.version>
+        <hazelcast-jet.version>5.0-SNAPSHOT</hazelcast-jet.version>
```
Comment (olukas) [inline pom.xml]: We should rename `hazelcast-jet.version` to `hazelcast.version`

File: road-traffic-predictor/src/main/resources/log4j.properties [hunk 5]
```diff
@@ -1,9 +1,9 @@
 log4j.appender.stdout=org.apache.log4j.ConsoleAppender
 log4j.appender.stdout.Target=System.out
 log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
-log4j.appender.stdout.layout.ConversionPattern=%d{mm:ss,SSS} %c{1} %m%n
+log4j.appender.stdout.layout.ConversionPattern=%d{mm:ss,SSS} %m%n
 
-log4j.logger.com.hazelcast.jet=info
+log4j.logger.com.hazelcast.jet=debug
```
Comment (olukas) [inline road-traffic-predictor/src/main/resources/log4j.properties]: Is this change (and change for `log4j.appender.stdout.layout.ConversionPattern`) intended or it's leftover?

File: pom.xml [hunk 6]
```diff
@@ -4,29 +4,27 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.hazelcast.jet.demos</groupId>
-    <artifactId>hazelcast-jet-demos</artifactId>
-    <version>4.3</version>
+    <groupId>com.hazelcast.platform.demos</groupId>
+    <artifactId>hazelcast-platform-demos</artifactId>
+    <version>5.0</version>
```
Comment (olukas) [inline pom.xml]: This should be `5.0-SNAPSHOT` version (and the version should be updated in submodules)

File: pom.xml [hunk 7]
```diff
@@ -4,29 +4,27 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.hazelcast.jet.demos</groupId>
-    <artifactId>hazelcast-jet-demos</artifactId>
-    <version>4.3</version>
+    <groupId>com.hazelcast.platform.demos</groupId>
```
Comment (olukas) [inline pom.xml]: I don't think `platform` should be used in `groupId` and `artifactId`. This makes more sense:
```
groupId=com.hazelcast.demos
artifactId=hazelcast-demos
```
@vladoschreiner WDYT?

File: pom.xml [hunk 8]
```diff
@@ -4,29 +4,27 @@
...
         <module>road-traffic-predictor</module>
-        <module>train-track</module>
         <module>tensorflow</module>
         <module>h2o-breast-cancer-classification</module>
-        <module>debezium-cdc-with-kafka</module>
-        <module>debezium-cdc-without-kafka</module>
     </modules>
 
     <repositories>
+        <repository>
```
Comment (olukas) [inline pom.xml]: Is this repository actually needed?

**Review by olukas — CHANGES_REQUESTED** [review]
@slavagithub please see inline comments.

Regarding `JetInstance` and `Hazelcast.bootstrappedInstance()` - if I understand it correctly, `Hazelcast.bootstrappedInstance()` should be replacement for `Jet.bootstrappedInstance()` but we still should use `JetInstance` for working with Jet, i.e. something like `JetInstance jet = Hazelcast.bootstrappedInstance().getJetInstance();`. 

And this is not replacement for `Jet.newJetInstance()`, we should use `Hazelcast.newHazelcastInstance().getJetInstance()` as replacement for it.

@ufukyilmaz can you please confirm that I understand it correctly? (AFAIK you're currently working on `Hazelcast.bootstrappedInstance()`)

**Comment by ufukyilmaz** [comment]
> @slavagithub please see inline comments.
> 
> Regarding `JetInstance` and `Hazelcast.bootstrappedInstance()` - if I understand it correctly, `Hazelcast.bootstrappedInstance()` should be replacement for `Jet.bootstrappedInstance()` but we still should use `JetInstance` for working with Jet, i.e. something like `JetInstance jet = Hazelcast.bootstrappedInstance().getJetInstance();`.
> 
> And this is not replacement for `Jet.newJetInstance()`, we should use `Hazelcast.newHazelcastInstance().getJetInstance()` as replacement for it.
> 
> @ufukyilmaz can you please confirm that I understand it correctly? (AFAIK you're currently working on `Hazelcast.bootstrappedInstance()`)

Hi @olukas , All what you said is true. To add to this discussion, we deprecate `Jet` class and all of the static factory methods of it (Jet was the main entry point for old Jet product), because we wanted the entry point of this merged product to be the `Hazelcast` class. Users will get `HazelcastInstance` using the methods of this Hazelcast class. Then, if they want to access `JetInstance`, they can get it only from `HazelcastInstance` by calling the `HazelcastInstance#getJetInstance()`. This will be only way to access `JetInstance` unless they use deprecated methods.

To list all the replacements:

- `Jet.newJetInstance()` should be replaced by `Hazelcast.newHazelcastInstance().getJetInstance()`
- `Jet.newJetClient()` -> `HazelcastClient.newHazelcastClient().getJetInstance()`
- `Jet.bootstrappedInstance()` or `JetBootstrap.getJetInstance()`(which is a usage of the internal API that we are removing in 5.0) -> `Hazelcast.bootstrappedInstance().getJetInstance()`
- `Jet.newJetFailoverClient()` -> `HazelcastClient.newHazelcastFailoverClient()`

**Comment by ufukyilmaz** [comment]
btw, https://github.com/hazelcast/hazelcast/pull/18672 including `Hazelcast.bootstrappedInstance()` changes, is not merged yet. It will be merged very soon.

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java (hunk 3 above)
Comment (slavagithub) [reply flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java]: This is done

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java (hunk 2 above)
Comment (slavagithub) [reply flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java:165]: It seems this is a common approach for "finally". It is in HZ code and Idea default formatted puts it into that way. Could we please discuss common approach?

File: pom.xml (hunk 4 above)
Comment (slavagithub) [reply pom.xml]: This is done

File: road-traffic-predictor/src/main/resources/log4j.properties (hunk 5 above)
Comment (slavagithub) [reply road-traffic-predictor/src/main/resources/log4j.properties]: leftover, removed :)

File: pom.xml (hunk 6 above)
Comment (slavagithub) [reply pom.xml]: It's done

File: pom.xml (hunk 7 above)
Comment (slavagithub) [reply pom.xml]: It seems this is a good idea, since we not a platform anymore)

File: pom.xml (hunk 8 above)
Comment (slavagithub) [reply pom.xml]: I don't think so. Removed

File: cryptocurrency-sentiment-analysis/README.md (hunk 1 above)
Comment (slavagithub) [reply cryptocurrency-sentiment-analysis/README.md]: added, but changes will go in separate PR - there is some work with Twitter Hazelcast library

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java (hunk 2 above)
Comment (olukas) [reply flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java:165]: It is definitely not common approach at all. If Idea put it to the same line I guess it's a bug in Idea. Ending `}` should be on the new line, I have never seen code like this:
```
        } finally {
            Hazelcast.shutdownAll();        }
```
It should be:
```
        } finally {
            Hazelcast.shutdownAll();
        }
```

However, this indicate we should apply some checkstyles rules to `hazelcast-demos` repo. I tried to run
```
grep "\;.*  }" -R . | grep "\.java:" | wc -l
``` 
and it give 47 results for `hazelcast-demos` repo. Things like this should not occur in our code.

File: cryptocurrency-sentiment-analysis/src/main/resources/twitter-security.properties [hunk 9]
```diff
@@ -1,4 +0,0 @@
-consumerKey=REPLACE_THIS
```
Comment (olukas) [inline cryptocurrency-sentiment-analysis/src/main/resources/twitter-security.properties:1]: I guess this file should not be removed, right?

**Review by olukas — APPROVED** [review]
LGTM thanks. Please fix mentioned checkstyle issue before merge and check whether removing of `cryptocurrency-sentiment-analysis/src/main/resources/twitter-security.properties` is expected.

File: flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java (hunk 2 above)
Comment (slavagithub) [reply flight-telemetry/src/main/java/com/hazelcast/jet/demo/FlightTelemetry.java:165]: yes, this example is 100% correct, will refactor where possible

File: cryptocurrency-sentiment-analysis/src/main/resources/twitter-security.properties (hunk 9 above)
Comment (slavagithub) [reply cryptocurrency-sentiment-analysis/src/main/resources/twitter-security.properties:1]: it should be with REPLACE_THIS for all keys

**Pull request merged**